custom_metrics/a11y.js

```diff
@@ -210,4 +210,184 @@ return JSON.stringify({
+
+      for (let source of node.name.sources || []) {
+        // Only include sources that contributed a value
+        if (!source.value || !source.value.value) {
+          continue;
+        }
+
+        // Only keep the relevant properties
+        const cleaned_source = {
+          type: source.type,
+          value: source.value.value,
+        };
+        if (source.attribute) {
+          cleaned_source.attribute = source.attribute;
+        }
+
+        control_stats.accessible_name_sources.push(cleaned_source);
+      }
+
+      accumulator.push(control_stats);
+    }
+
+    const allowed_control_types = [
+      'input',
+      'select',
+      'textarea',
+      'button',
+    ];
+
+    const stats_of_controls = [];
+    for (const node of $WPT_ACCESSIBILITY_TREE) {
+      const node_type = (node.node_info?.nodeType || '').toLowerCase();
+      if (!allowed_control_types.includes(node_type)) {
+        continue;
+      }
+
+      addControlToStats(node, stats_of_controls);
+    }
+
+    return stats_of_controls;
+  }),
+  // If the radio or checkbox elements are inside a fieldset or legend
+  fieldset_radio_checkbox: captureAndLogError(() => {
+    let total_radio_in_fieldset = 0;
+    let total_checkbox_in_fieldset = 0;
+    const fieldset_stats = [];
+
+    const fieldset_elements = document.querySelectorAll('fieldset');
+    for (let fieldset of fieldset_elements) {
+      let has_legend = !!fieldset.querySelector('legend');
+      const total_radio = fieldset.querySelectorAll('input[type="radio"]').length
+      const total_checkbox = fieldset.querySelectorAll('input[type="checkbox"]').length
+
+      total_radio_in_fieldset += total_radio;
+      total_checkbox_in_fieldset += total_checkbox;
+
+      fieldset_stats.push({
+        has_legend,
+        total_radio,
+        total_checkbox,
+      });
+    }
+
+    return {
+      total_radio: document.querySelectorAll('input[type="radio"]').length,
+      total_checkbox: document.querySelectorAll('input[type="checkbox"]').length,
+      total_radio_in_fieldset,
+      total_checkbox_in_fieldset,
+
+      fieldsets: fieldset_stats,
+    }
+  }),
+  required_form_controls: captureAndLogError(() => {
+    function getVisibleLabel(element) {
+      // Explicit label
+      const id = (element.getAttribute('id') || '').trim();
+      if (id.length > 0) {
+        const element = document.querySelector(`label[for="${id}"]`);
+        if (element) {
+          return element.textContent.trim();
+        }
+      }
+
+      // Implicit label
+      if (element.parentElement && element.parentElement.tagName === 'LABEL') {
+        return element.parentElement.textContent.trim();
+      }
+
+      return null;
+    }
+
+    function hasRequiredAsterisk(element) {
+      const label = getVisibleLabel(element);
+      if (!label) {
+        return false;
+      }
+
+      if (label.substr(0, 1) === '*' || label.substr(-1, 1) === '*') {
+        return true;
+      }
+
+      return false;
```

Comment:
```suggestion
      return label.startsWith('*') || label.endsWith('*');
```

Reply:
Always forget about the "new" methods. Even though they've been out for a while now